dali/pipeline/operator/builtin/external_source.h

```diff
@@ -123,20 +122,17 @@ class ExternalSource : public Operator<Backend> {
     return "ExternalSource (" + output_name_ + ")";
   }
 
-
-  /**
-   * @brief Sets the data that should be passed out of the op
-   * on the next iteration.
-   */
-  template<typename SrcBackend>
-  inline void SetDataSource(const TensorList<SrcBackend> &tl, cudaStream_t stream = 0) {
+  template<typename SrcBackend, template<typename> class SourceDataType>
+  inline void SetDataSourceHelper(const SourceDataType<SrcBackend> &t, cudaStream_t stream = 0) {
     if (std::is_same<SrcBackend, GPUBackend>::value && std::is_same<Backend, CPUBackend>::value) {
-      std::string msg = "Incorrect Backends warning. Loading GPU-originated data into CPU "
-                        "ExternalSource operator is discouraged and might be inefficient.";
-      DALI_WARN(msg);
+      DALI_WARN("Incorrect Backends warning. Loading GPU-originated data into CPU " +
+                "ExternalSource operator is discouraged and might be inefficient.");
```

Comment:
There's no point in adding strings this way - subsequent literals will be merged at compile time.

Reply:
Done